static_website/dispatch.ml

```diff
@@ -1,53 +1,50 @@
 open Lwt.Infix
-open V1_LWT
 
-module Main (C:CONSOLE) (FS:KV_RO) (S:Cohttp_lwt.Server) = struct
+let server_src = Logs.Src.create "server" ~doc:"HTTP server"
+module Server_log = (val Logs.src_log server_src : Logs.LOG)
 
-  let start c fs http =
+module Main (Clock:V1.CLOCK) (FS:V1_LWT.KV_RO) (S:Cohttp_lwt.Server) = struct
+  module Logs_reporter = Mirage_logs.Make(Clock)
 
-    let read_fs name =
-      FS.size fs name >>= function
+  let read_fs fs name =
+    FS.size fs name >>= function
+    | `Error (FS.Unknown_key _) -> Lwt.fail (Failure ("read " ^ name))
```

Comment:
It will have exactly the same effect here as `raise`: an exception is returned to the caller. In this case, the caller explicitly silently discards exceptions.